docs/architecture/adr-074-timeout-params.md

```diff
@@ -0,0 +1,228 @@
+* `TimeoutCommit` 
+	* How long the consensus algorithm waits after committing a block but before starting the new height.
+	* This gives a validator a chance to receive slow precommits.
+* `SkipTimeoutCommit`
+	* Make progress as soon as the node has 100% of the precommits.
+
+
+### Overview of Proposed Change
+
+This ADR proposes consolidating the timeout parameters and migrating them from the
+node-local `config.toml` file into the network-global consensus parameters.
+The consensus parameters are the same across all validators on the network.
+
+A sane default will be provided by Tendermint for each of these parameters and
+networks will be able to update the parameters as they see fit. Local updates
+to these parameters will no longer be possible; instead, the application will control
+updating the parameters. Applications using the Cosmos SDK will be automatically be
+able to change the values of these consensus parameters [via a governance proposal][cosmos-sdk-consensus-params].
+This ADR also proposes consolidating the 8 timeout parameters into 6. These
+6 parameters would be as follows:
+
+* `TimeoutPropose`
+	* Same as current `TimeoutPropose`.
+* `TimeoutProposeDelta`
+	* Same as current `TimeoutProposeDelta`.
+* `TimeoutVote`
+	* Parameter that governs how long validators wait for votes in both the prevote
+	 and precommit phase of the consensus algorithm. This parameter will subsume
+	 the current `TimeoutPrevote` and `TimeoutPrecommit` parameters.
+* `TimeoutVoteDelta`
+	* Parameter that governs how much the `TimeoutVote` will grow each successive round.
+	 This parameter will subsume the current `TimeoutPrevoteDelta` and `TimeoutPrecommitDelta`
+	 parameters.
+* `TimeoutCommit`
+	* Same as current `TimeoutCommit`.
+* `SkipTimeoutCommit`
+	* Same as current `SkipTimeoutCommit`.
+
+This change is low risk. While these values may be configured, many running chains
+do not change these parameters from their default values. For example, initializing
+a node on Osmosis, Terra, and the Cosmos Hub using the their `init` command produces
+a `config.toml` with Tendermint's default values for these parameters.
+
+### Why this parameter consolidation?
+
+Reducing the number of parameters is good for UX. Fewer superfluous parameters makes
+running and operating a Tendermint network less confusing. 
+
+The Prevote and Precommit messages are both similar sizes and are also similar in
+their function in the consensus algorithm so there is no strong need for them to
+be configured separately.
+
+The `TimeoutPropose` parameter governs how long Tendermint will wait for the proposed
+block to be gossiped. Blocks are much larger than votes and therefore tend to be
+gossiped much more slowly. It therefore makes sense to keep `TimeoutPropose` and
+the `TimeoutProposeDelta` as parameters separate from the vote timeouts.
+
+`TimeoutCommit` is used by chains to ensure that the network waits for the votes from
```

Comment:
> For example, say I run a cabal of fast validators on good network, with near or over 2/3 voting power. How am I enforced against patching my validators to skip the block delay and start the next round immediately? In this sense, my validators are Byzantine—they violate the protocol—but they do so in a way that's hard to detect.

There is nothing that stops this and we do not currently generate evidence about this behavior. Here are the types of evidence we do collect:
https://github.com/tendermint/tendermint/blob/2c55eead7c71f4686711bcec198de8bd7a3d4714/proto/tendermint/types/evidence.pb.go#L76

This would be difficult evidence to collect because you would need proof that a validator collected the precommits from a height and executed the associated block _later_ than it claims to have done so but there is no trusted way currently to gather information about when a validator gathered its data. The best I believe you would be able to do would be determine if a validator issued a proposal/prevote/precommit before the time `2/3rth_precommit_issued + timeout_commit` and generate evidence to prevent egregiously bad behavior.

I agree that it would be a byzantine behavior and it would confer some degree of economic advantage to the fast validators: they would be able to freeze out the other validators as proposers and therefore collected the block rewards for proposing.